plugins/user-authenticators/saml2/src/main/java/org/apache/cloudstack/saml/SAMLUtils.java

```diff
@@ -150,7 +150,7 @@ public static String buildAuthnRequestUrl(final String authnId, final SAMLProvid
             if (spMetadata.getKeyPair() != null) {
                 privateKey = spMetadata.getKeyPair().getPrivate();
             }
-            redirectUrl = idpMetadata.getSsoUrl() + "?" + SAMLUtils.generateSAMLRequestSignature("SAMLRequest=" + SAMLUtils.encodeSAMLRequest(authnRequest), privateKey, signatureAlgorithm);
+            redirectUrl = idpMetadata.getSsoUrl().contains("?") ? idpMetadata.getSsoUrl() + "&" + SAMLUtils.generateSAMLRequestSignature("SAMLRequest=" + SAMLUtils.encodeSAMLRequest(authnRequest), privateKey, signatureAlgorithm) : idpMetadata.getSsoUrl() + "?" + SAMLUtils.generateSAMLRequestSignature("SAMLRequest=" + SAMLUtils.encodeSAMLRequest(authnRequest), privateKey, signatureAlgorithm);
```

Comment:
for readibility :
```suggestion
            String appendOperator = idpMetadata.getSsoUrl().contains("?") ? "&" : "?";
            redirectUrl = idpMetadata.getSsoUrl()
                 + appendOperator
                 + SAMLUtils.generateSAMLRequestSignature("SAMLRequest="
                     + SAMLUtils.encodeSAMLRequest(authnRequest), privateKey, signatureAlgorithm);
```

Reply:
Addressed https://github.com/apache/cloudstack/pull/6457/commits/749dca23b7fe86c4dbf811c1afe14ef5f444f413